File: app/src/main/java/io/givenow/app/activities/SplashActivity.java
package io.givenow.app.activities;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;

/**
 * Created by aphex on 11/23/15.
 */
public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Intent targetIntent;
        if (isFirstTime()) {
            targetIntent = new Intent(this, OnboardingActivity.class);
        } else {
            targetIntent = new Intent(this, MainActivity.class);
        }
        // So push notification data gets passed on:
        if (getIntent() != null) {
            if (getIntent().getExtras() != null) {
                targetIntent.putExtras(getIntent().getExtras());
            }
        }

        startActivity(targetIntent);
        finish();
    }

    /***
     * Checks that application runs first time and write flag at SharedPreferences
     *
     * @return true if 1st time
     */
    private boolean isFirstTime() {
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getBaseContext());
        boolean ranBefore = preferences.getBoolean("RanBefore", false);
        return !ranBefore;
    }
}

File: app/src/main/java/io/givenow/app/activities/SplashActivity.kt
package io.givenow.app.activities

import android.content.Intent
import android.os.Bundle
import android.preference.PreferenceManager
import android.support.v7.app.AppCompatActivity

/**
 * Created by aphex on 11/23/15.
 *
 */
class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val targetIntent: Intent = when {
            isFirstTime -> Intent(this, OnboardingActivity::class.java)
            else -> Intent(this, MainActivity::class.java)
        }

        // So push notification data gets passed on:
        intent?.extras?.let { extras -> targetIntent.putExtras(extras) }
//        if (intent != null) {
//            if (intent.extras != null) {
//                targetIntent.putExtras(intent.extras)
//            }
//        }

        startActivity(targetIntent)
        finish()
    }

    /***
     * Checks that application runs first time and write flag at SharedPreferences

     * @return true if 1st time
     */
    private val isFirstTime: Boolean
        get() {
            val ranBefore = PreferenceManager.getDefaultSharedPreferences(baseContext).getBoolean("RanBefore", false)
            return !ranBefore
        }
}
